io.openems.edge.io.shelly/src/io/openems/edge/io/shelly/shellyplus3em/IoShellyPlus3emImpl.java

```diff
@@ -0,0 +1,266 @@
+import io.openems.edge.common.component.AbstractOpenemsComponent;
+import io.openems.edge.common.component.OpenemsComponent;
+import io.openems.edge.common.event.EdgeEventConstants;
+import io.openems.edge.io.api.DigitalOutput;
+import io.openems.edge.meter.api.ElectricityMeter;
+import io.openems.edge.meter.api.MeterType;
+import io.openems.edge.meter.api.SinglePhase;
+import io.openems.edge.meter.api.SinglePhaseMeter;
+import io.openems.edge.timedata.api.Timedata;
+import io.openems.edge.timedata.api.TimedataProvider;
+import io.openems.edge.timedata.api.utils.CalculateEnergyFromPower;
+
+@Designate(ocd = Config.class, factory = true)
+@Component(
+        name = "IO.Shelly.Plus3EM",
+        immediate = true,
+        configurationPolicy = ConfigurationPolicy.REQUIRE
+)
+@EventTopics({
+        EdgeEventConstants.TOPIC_CYCLE_EXECUTE_WRITE,
+        EdgeEventConstants.TOPIC_CYCLE_AFTER_PROCESS_IMAGE
+})
+public class IoShellyPlus3emImpl extends AbstractOpenemsComponent implements IoShellyPlus3em, DigitalOutput,
+        SinglePhaseMeter, ElectricityMeter, OpenemsComponent, TimedataProvider, EventHandler {
+
+	private final CalculateEnergyFromPower calculateProductionEnergy = new CalculateEnergyFromPower(this,
+			ElectricityMeter.ChannelId.ACTIVE_PRODUCTION_ENERGY);
+	private final CalculateEnergyFromPower calculateConsumptionEnergy = new CalculateEnergyFromPower(this,
+			ElectricityMeter.ChannelId.ACTIVE_CONSUMPTION_ENERGY);
+
+	
+    private final Logger log = LoggerFactory.getLogger(IoShellyPlus3emImpl.class);
+    private final BooleanWriteChannel[] digitalOutputChannels;
+
+    private MeterType meterType = null;
+    private SinglePhase phase = null;
+    private String baseUrl;
+
+	private volatile Timedata timedata;
+    
+    @Reference(cardinality = ReferenceCardinality.MANDATORY)
+    private BridgeHttpFactory httpBridgeFactory;
+    private BridgeHttp httpBridge;
+
+    public IoShellyPlus3emImpl() {
+        super(
+                OpenemsComponent.ChannelId.values(),
+                ElectricityMeter.ChannelId.values(),
+                DigitalOutput.ChannelId.values(),
+                IoShellyPlus3em.ChannelId.values()
+        );
+        this.digitalOutputChannels = new BooleanWriteChannel[]{
+                this.channel(IoShellyPlus3em.ChannelId.RELAY)
+        };
+
+        ElectricityMeter.calculateSumActivePowerFromPhases(this);
+    }
+
+    @Activate
+    protected void activate(ComponentContext context, Config config) {
+        super.activate(context, config.id(), config.alias(), config.enabled());
+        this.meterType = config.type();
+        this.baseUrl = "http://" + config.ip();
+        this.httpBridge = this.httpBridgeFactory.get();
+
+        if (this.isEnabled()) {
+            this.httpBridge.subscribeJsonEveryCycle(this.baseUrl + "/status", this::processHttpResult);
+        }
+    }
+
+    @Deactivate
+    protected void deactivate() {
+        if (this.httpBridge != null) {
```

Comment:
you dont need to check if the `httpBridge` is null the `unget` method allows null values and ignores them